File: Greedy/Jump Game 2/Solution.py
class Solution:
    def jump(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        cur_max = next_max = 0
        step = 0
        for ind, val in enumerate(nums):
            if ind > cur_max:
                step += 1
                cur_max = next_max
            next_max = max(next_max, ind + val)
            # todo: 到达终点即停止
        return step

File: Greedy/Jump Game 2/Solution.cpp
Class Solution{
public:
    int jump(vector<int>& nums){
        int local = 0, global = 0, step = 0;
        for (int i = 0; i < nums.size(); i++){
            if (i > local){
                local = global;
                step++;
            }
            global = max(global, i + nums[i]);
            # todo: 到达终点即停止
        }
        return step;
    }
}